src/functions/asyncCredential/sessionService/sessionService.ts

```diff
@@ -74,14 +119,41 @@ export class SessionService implements IRecoverAuthSession {
       result.Items[0].sessionId.S !== ""
     );
   }
+
+  private async checkSessionsExists(authSessionId: string): Promise<boolean> {
+    const output = await dbClient.send(
+      new GetItemCommand({
+        TableName: this.tableName,
+        Key: {
+          sessionId: { S: authSessionId },
+        },
+      }),
+    );
+
+    return output.Item != null;
+  }
+}
+
+interface ISessionConfig {
+  authSessionId: string;
+  state: string;
+  sub: string;
+  client_id: string;
+  govuk_signin_journey_id: string;
+  redirect_uri: string;
+  issuer: string;
+  sessionState: string;
 }
 
-export interface IRecoverAuthSession {
+export interface ISessionService {
   getAuthSessionBySub: (
     sub: string,
     state: string,
     sessionRecoveryTimeout: number,
   ) => Promise<ErrorOrSuccess<string | null>>;
+  createSession: (
+    sessionConfig: ISessionConfig,
+  ) => Promise<ErrorOrSuccess<null>>;
```

Comment:
FYI, this will eventually return a sessionId so that we can include it in our COMPLETED log, which is yet to be written (and out of scope for this change)

Reply:
Isn't the sessionId generated in the lambda so we don't need to return it?

Reply:
We do return the sessionId in the above function used for recovering a session